generator.js

```diff
@@ -0,0 +1,40 @@
+const fs = require('fs')
+const faker = require('faker')
+const argv = require('yargs').argv
+const helpers = require('./helpers/helpers.js')
+
+const residences = argv.residences || 10000000
+const filename = argv.output || 'residences.csv'
+const stream = fs.createWriteStream(filename)
+
+const createResidence = () => {
+  const residenceName = faker.company.companyName();
+  const price = faker.commerce.price();
+  const url = helpers.generateRandomImage();
+  return `${residenceName},${price},${url}\n`
+}
+
+const startWriting = (writeStream, encoding, done) => {
+  let i = residences;
+  function writing() {
+    let canWrite = true
+    do {
+      i--
+      let residence = createResidence();
+      if (i === 0) {
+        writeStream.write(residence, encoding, done)
+      } else {
+        writeStream.write(residence, encoding)
+      }
+    } while (i > 0 && canWrite)
+      if (i > 0 && !canWrite) {
```

Comment:
This is the one part in here I think I can't follow. Is 'canWrite' a property of your buffer stream that you set to true by default but will get flipped back to false if the buffer is, in fact, full and we can't write?